### PR TITLE
Using Hotstar without valid login (Not for VIP or Premium content)

### PR DESCRIPTION
**Pull request by Vikassm73**
You will able to use this app without valid login details. First you have to click "Login Free" from the setting before entering the addon. Once done, play any video. First play gives an issue (Need fixing). Just return from playing and start playing another video.There after it work smoothly.

### CODE REVIEW

File: plugin.video.botallen.hotstar/resources/settings.xml [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <category label="General"><!--General-->
   <setting label="Account" type="lsep"/><!--Account-->
   <setting id="login" type="action" label="Login" action="RunPlugin(plugin://plugin.video.botallen.hotstar/resources/lib/main/login/)" option="close"/>
+  <setting id="loginfree" type="action" label="Login Free" action="RunPlugin(plugin://plugin.video.botallen.hotstar/resources/lib/main/loginfree/)" option="close"/>
```
Comment (botallen) [inline plugin.video.botallen.hotstar/resources/settings.xml:6]:
```suggestion
  <setting id="loginfree" type="action" label="Login Free" action="RunPlugin(plugin://plugin.video.botallen.hotstar/resources/lib/main/loginfree/)" option="close"/>
```

This option is unnecessary.

File: plugin.video.botallen.hotstar/resources/lib/main.py [hunk 2]
```diff
@@ -69,6 +69,10 @@ def login(plugin):
         pdialog.update(i, line3="[B][UPPERCASE]%s[/UPPERCASE][/B]" % code)
     pdialog.close()
 
+@Script.register
```
Comment (botallen) [inline plugin.video.botallen.hotstar/resources/lib/main.py:72]:
```suggestion
@Script.register
```

Also this is not required

**Comment by botallen** [comment]
Also you have to handle logged in error explicitly [here](https://github.com/botallen/repository.botallen/blob/9895482b461cb5dc2db336ded0398129768431fc/plugin.video.botallen.hotstar/resources/lib/api.py#L128). Because isLoggedIn function will allow to play but if user is not logged in then hotstar will give error with some response code and on that particular response we can prompt user for login

**Comment by Vikassm73** [comment]
I tried this method. This way we can skip additional tab in the setting but it gives an error in the first playback. Then again additional changes in code is required to tackle this error. I have added self._refreshToken() in the free login code. now everything is working fine.

**Comment by botallen** [comment]
What if the stream is not for guest and requires login.

**Comment by Vikassm73** [comment]
if stream is not for the guest then it will raise subscription error. In that case, user will have to logout and do valid login using Login option.

**Pull request closed without merging**